stdlib/public/SwiftShims/System.h

```diff
@@ -152,10 +152,19 @@
 /// Darwin reserves the low 4GB of address space.
 #define SWIFT_ABI_DARWIN_ARM64_LEAST_VALID_POINTER 0x100000000ULL
 
+// Android AArch64 reserves the top byte for pointer tagging since Android 11,
+// so shift these tags to the second byte.
+#define SWIFT_ABI_ANDROID_ARM64_SWIFT_SPARE_BITS_MASK 0x00F0000000000007ULL
+#define SWIFT_ABI_ANDROID_ARM64_OBJC_RESERVED_BITS_MASK 0x0080000000000000ULL
```

Comment:
I only moved the spare bits mask initially, as this ObjC mask appeared to have no effect, but after turning stdlib assertions on to test the new `_internalInvariantFailure` assertions below, this mask being in the top byte caused several tests to fail, because of the `isNative` checks some assertions added. Because of those stdlib assertion failures alone, I shifted this ObjC mask and all stdlib tests then passed in Debug+assert mode.

Reply:
What happens if we just omit this definition then? That is, does it have less source impact to omit objc masks and support?

Reply:
This OBJC mask is completely unused except in some assertions, which simply check `isNative` in some invariants, even though that makes no sense with ObjC interoperability turned off? Rather than have all those tests fail when assertions are turned on, I added this mask.

However, I don't really care if this assertions workaround is upstreamed and can remove it if you like.

Reply:
The problem is that without this change [the default is to reserve the top bit on all AArch64 platforms](https://github.com/apple/swift/blob/0ab3eec987aefadf7dd97740d7baaa785005edb9/stdlib/public/SwiftShims/System.h#L161), which is then [checked against the top bit in Android's memory tag in `isNative`](https://github.com/apple/swift/blob/abec55f4324c98ef69b94100b649c83bdbfc8f91/stdlib/public/core/BridgeStorage.swift#L89), and `isNative` is checked by several assertions.

Presumably this is a mistake in checking this ObjC bit even when ObjC interoperability is turned off, so I worked around it by moving the ObjC bit down a byte, out of the Android memory tag in the top byte.

If you'd rather set this mask to zero everywhere on Android AArch64, that might work too.

Reply:
I set this OBJC mask to zero and ran the full compiler validation suite on a Debug+asserts stdlib and got the same passing test results, so I've now set this mask to zero. Since this is just a workaround for stdlib assertions that wrongly check `isNative` even when ObjC interop is turned off, I'm fine with not modifying this OBJC mask in any way in this pull and fixing that root cause in a separate pull later instead.

@milseman, just let me know which you prefer.